### ISSUE TITLE
Comments on the updated draft

### ISSUE THREAD

**Issue by m4mb01t4l14n0** [DavidSchinazi/draft-schinazi-httpbis-transport-auth#15]
Thank you for the draft updates, @DavidSchinazi.  Here's a shot at a "validating scenario" for HTTP Unprompted Auth:

We seek to emulate the phenomenon of the "speakeasy" in Prohibition America, with a bit more rigor to support higher levels of usage.  

Imagine a back alley deep in a maze of warehouse buildings in 1920s Chicago USA.  A person stands in front of door beneath a dimly lit bulb. Another person arrives and says "Is this the club where they serve free beer?".  The person answers "I don't know what you mean" and the man leaves. Another person arrives later and, without being asked, nods their head to the side and says "Whiskey Tango Foxtrot".  The person is let in the door.  A third person, arriving much later, emulates the second person but is told "I don't know what you mean".  

The second person in this scenario knows the speakeasy's location, knows the code and knows to provide it without being asked.   Everyone else fails to gain access and learns no information about the failure.  As a secondary measure, anyone who re-uses a previously-used code similarly fails to gain access and learns no information about the failure.  This secondary measure assures the code itself can not be shared along with the location. 

In an HTTP use case, certain resources provided by an origin require a mechanism for differentiating valid users from invalid ones. Validity is determined by perhaps account settings, expressed preferences or fees. These resources are valuable and, if the origin was known, it could be flooded with invalid requests, rendering the resources unusable by all.  Therefore the resource location is provided only to known valid users.  However, to prevent unwanted sharing of the location with invalid users, a mechanism is necessary to allow valid users to gain entry in a manner that does not indicate (a) the resource's location has been found or (b) that a special authentication is required.  The mechanism proposed by HTTP Unprompted Authentication supports this scenario by (a) using per-connection private data combined with pre-agreed or pre-generated authentication tokens and (b) generic failure codes.

**Comment by m4mb01t4l14n0** [comment]
If it's useful to present "real world use cases" and further if you think it's safe to provide Internet Freedom use cases, I think VPN access and configuration are good examples.  Someone at IETF115 mentioned access to Tor bridges. There are several different mechanisms where it would be useful just looking at Tor.  Pluggable Transports - which require another level of proxying, and are not specific to Tor - are another example.  Realistically, all the VPN providers need to wrap what are in reality a small number of network tunnel technologies (OpenVPN, Wireguard, Shadowsocks, etc) in configuration and access techniques that discriminate valid from invalid users.  And as they are, in many cases, under attack, anti-probing is very useful.

**Comment by m4mb01t4l14n0** [comment]
I recall that key exporters require that both sides use the same bit pattern when requesting the export (which serves to assure the same output hash is provided to both).  That's just a text string and in our case is the string name of the authentication scheme used.   Thus Signature and HMAC will generate two different hashes on output.  Should we indicate specifically that, as long as both sides agree in advance (the case of a "captive" client and server, as with VPN use cases) then in reality ANY agreed string can be used, in addition to the offically-registered schemes?

**Comment by DavidSchinazi** [comment]
Regarding the validating scenario, I think that having that example is really helpful in the discussion of whether the HTTPBIS WG should adopt the draft, but maybe it might be best to avoid it in the draft to reduce the odds of folks trying to block this.

For the key exporter, what's the benefit of using another string? Conceptually it's another shared secret which would be redundant with the keys we already have?

**Comment by m4mb01t4l14n0** [comment]
Yes, the scenario was for verbal discussion (I can't remember who said "we need a good use case"). 

Re the key exporter: I remember when you call the OpenSSL function you provide a string as input and it's hashed with the session key to give you a new exported key.  So, that string matters - both sides have to use the same string to extract the same value and this draft proposes to use the name of the authentication type being used.  It makes sense to register the two auth schemes defined in the draft, but in Internet Freedom use cases, others might want to use different auth schemes "privately" (that is, the application authors define a scheme name and an auth process that they use for their own app.  I'm just thinking we should highlight the fact that this is possible, or maybe even a feature.  E.g., a vendor might develop a private Zero Knowledge authentication scheme that requires the vendor to have very little actual identifying information about the user.  I'm talking about a half-way point between something like Tor that doesn't require any user authentication and a VPN that has (possibly billed) customers with emails or phone numbers.

**Comment by DavidSchinazi** [comment]
I still fail to see why changing the string accomplishes anything? Why not change the key instead? This schemes allows you to use the same key for all users if you'd like.
